src/chat-bubble/chat-bubble-base.ts

```diff
@@ -0,0 +1,68 @@
+import { html, LitElement, nothing } from "lit";
+import { property } from "lit/decorators.js";
+import { logger } from "../utils";
+import { AUTHORS, BaseParts, BaseSlots } from "./constants";
+
+export class ChatBubbleBase extends LitElement {
+  @property({ type: String, reflect: true })
+  public author!: (typeof AUTHORS)[number];
+
+  @property({ type: String, reflect: true })
+  public timestamp!: string;
+
+  @property({ type: Boolean, reflect: true, attribute: "fully-rounded" })
+  public fullyRounded: boolean = false;
+
+  constructor() {
+    super();
+  }
+
+  private _renderFooter() {
+    if (!this.timestamp) {
+      logger(
+        `Expected valid \`timestamp\` prop. received: \`${this.timestamp}\`.`,
+        "ChatBubble",
+        "error",
+      );
+
+      return nothing;
+    }
+
+    return html`
+      <div
+        class=${BaseParts.FOOTER}
+        part=${BaseParts.FOOTER}
+      >
+        <slot name=${BaseSlots.FOOTER}></slot>
+        <span>${this.timestamp}</span>
+      </div>
+    `;
+  }
+
+  protected override render() {
+    if (!AUTHORS.includes(this.author)) {
```

Comment:
Shouldn't we delegate this type of validation to the type checker?

Reply:
The type checker does not enforce our mandatory properties. If you look closely, you'll see the ! in front of the mandatory/required properties, indicating they are required. However, from the end-user's perspective (the developer using this element), no alerts will show if they miss a mandatory property. For reference, consider Web Standard HTML tags: most require mandatory attributes, but we do not get any alerts if these are missed.